--- a/buildtools/wafsamba/samba_third_party.py
+++ b/buildtools/wafsamba/samba_third_party.py
@@ -49,17 +49,10 @@ Build.BuildContext.CHECK_PAM_WRAPPER = CHECK_PAM_WRAPPER
 
 @conf
 def CHECK_LIBQUIC(conf):
-    return conf.CHECK_BUNDLED_SYSTEM_PKG('libquic', minversion='1.1')
+    return False
 Build.BuildContext.CHECK_LIBQUIC = CHECK_LIBQUIC
 
 @conf
 def CHECK_LIBNGTCP2(conf):
-    minversion = '1.12.0'
-    if not conf.CHECK_BUNDLED_SYSTEM_PKG('libngtcp2_crypto_gnutls',
-                                         minversion=minversion):
-        return False
-    if not conf.CHECK_BUNDLED_SYSTEM_PKG('libngtcp2',
-                                         minversion=minversion):
-        return False
-    return True
+    return False
 Build.BuildContext.CHECK_LIBNGTCP2 = CHECK_LIBNGTCP2
