--- a/contrib/backends/pagekitec.c
+++ b/contrib/backends/pagekitec.c
@@ -153,7 +153,7 @@ int summarize_status(const char* fpath,
     }
 
     fprintf(fd, format,
-      PK_VERSION, status_msg, status, getpid(), STATUS_MIN_INTERVAL, now);
+      PK_VERSION, status_msg, status, getpid(), STATUS_MIN_INTERVAL, (long long)now);
 
     if (0 == ftruncate(fileno(fd), ftell(fd))) fflush(fd);
     fclose(fd);
