--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -353,13 +353,6 @@
   #Use the homebrew version. MacOS provided ICU doesn't provide development files
   set(ICU_ROOT "/usr/local/opt/icu4c")
 endif()
-find_package(
-  ICU
-  COMPONENTS
-    i18n
-    uc
-    data
-)
 set_package_properties(
   ICU
   PROPERTIES
@@ -406,7 +399,6 @@
   endif()
 endif()
 set(BerkeleyDB_FIND_QUIETLY True)
-find_package(BerkeleyDB)
 set_package_properties(
   BerkeleyDB
   PROPERTIES
