--- a/src/TextClient.cpp
+++ b/src/TextClient.cpp
@@ -356,7 +356,7 @@
 							break;
 						} else if ( hash.Decode(token.Trim(false).Trim(true)) ) {
 							if ( !hash.IsEmpty() ) {
-								Show(_("Processing by hash: "+token+wxT("\n")));
+								Show(_("Processing by hash: ") + token + wxT("\n"));
 								request->AddTag(CECTag(EC_TAG_PARTFILE, hash));
 							}
 						} else {
@@ -370,7 +370,7 @@
 									partmetname == token ||
 									partmetname.Truncate(partmetname.Len()-4) == token ||
 									partmetname.Truncate(partmetname.Len()-5) == token) {
-									Show(_("Processing by filename: "+token+wxT("\n")));
+									Show(_("Processing by filename: ") + token + wxT("\n"));
 									request->AddTag(CECTag(EC_TAG_PARTFILE, tag->FileHash()));
 								}
 							}
