--- a/Cargo.toml
+++ b/Cargo.toml
@@ -61,7 +61,7 @@
 [features]
 alsa_backend = ["librespot-playback/alsa-backend", "dep:alsa"]
 dbus_mpris = ["dep:dbus", "dep:dbus-tokio", "dep:dbus-crossroads"]
-default = ["alsa_backend", "pulseaudio_backend", "dbus_mpris"]
+default = ["alsa_backend", "dbus_mpris"]
 portaudio_backend = ["librespot-playback/portaudio-backend"]
 pulseaudio_backend = ["librespot-playback/pulseaudio-backend"]
 rodio_backend = ["librespot-playback/rodio-backend"]
