--- a/src/installer/destinations.py
+++ b/src/installer/destinations.py
@@ -169,7 +169,7 @@ class SchemeDictionaryDestination(WheelDestination):
         target_path = self._path_with_destdir(scheme, path)
         if not self.overwrite_existing and target_path.exists():
             message = f"File already exists: {target_path!s}"
-            raise FileExistsError(message)
+            target_path.unlink()
 
         parent_folder = target_path.parent
         if not parent_folder.exists():
